src/server/object_services/md_store.js

```diff
@@ -456,20 +474,20 @@ class MDStore {
                 $lt: new Date(moment.unix(max_create_time).toISOString()),
                 $exists: true
             } : undefined,
+            size: size_filter,
             upload_started: typeof upload_mode === 'boolean' ? {
                 $exists: upload_mode
             } : undefined,
+            tagging,
```

Comment:
I'm not sure this does what you intend it to do. you want to find all objects that has all of the given tags, but I think that this will search for a perfect match between the query tags and the object tags

Reply:
Thanks for pointing this out. See [Postgres client change](https://github.com/noobaa/noobaa-core/pull/6919/files#diff-c020c466408e4072550dbc75a3b7144063b7162ce224cdac0d3ef0d253acd794), and added test with an out-of-order subset of tags match.

Reply:
shouldn't that be changed to:

```suggestion
            tagging: {
               $all: tagging
            },
```

Reply:
`$all` operator is applied [here](https://github.com/baum/noobaa-core/blob/45b242d714f4408b1c2558b40c1a8e04479d4a0a/src/server/object_services/object_server.js#L978-L980).

Reply:
I think we should have the DB related code in the md_store and not object_server

Reply:
Moved [here](https://github.com/baum/noobaa-core/blob/d068eea5c4ec7e54bcf19b656092203dd7672afe/src/server/object_services/md_store.js#L477-L479).